packages/flutter_driver/lib/src/extension/wait_conditions.dart

```diff
@@ -177,18 +177,12 @@ class _InternalCombinedCondition implements WaitCondition {
 /// Parses a [WaitCondition] or its subclass from the given serializable [waitCondition].
 WaitCondition deserializeCondition(SerializableWaitCondition waitCondition) {
   final String conditionName = waitCondition.conditionName;
-  switch (conditionName) {
-    case 'NoTransientCallbacksCondition':
-      return _InternalNoTransientCallbacksCondition.deserialize(waitCondition);
-    case 'NoPendingFrameCondition':
-      return _InternalNoPendingFrameCondition.deserialize(waitCondition);
-    case 'FirstFrameRasterizedCondition':
-      return _InternalFirstFrameRasterizedCondition.deserialize(waitCondition);
-    case 'NoPendingPlatformMessagesCondition':
-      return _InternalNoPendingPlatformMessagesCondition.deserialize(waitCondition);
-    case 'CombinedCondition':
-      return _InternalCombinedCondition.deserialize(waitCondition);
-  }
-  throw SerializationException(
-      'Unsupported wait condition $conditionName in ${waitCondition.serialize()}');
+  return switch (conditionName) {
```

Comment:
Is the variable `conditionName` still necessary?

Reply:
Nope! I just have it here because of the [line length guideline](https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#prefer-a-maximum-line-length-of-80-characters):

```dart
WaitCondition deserializeCondition(SerializableWaitCondition waitCondition) {
  return switch (waitCondition.conditionName) {
    // ...
    final String conditionName => throw SerializationException('Unsupported wait condition $conditionName in ${waitCondition.serialize()}'),
  };
```

Defining the variable beforehand lets us use a `_` instead.